### PR TITLE
IncludeFixer - remove braces when the statement is wrapped in block

### PR DESCRIPTION
**Pull request by kubawerlos**
Ping @umpirsky, @sstok and @keradus (as authors) for review.

### CODE REVIEW

File: tests/Tokenizer/Analyzer/BlocksAnalyzerTest.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        return [
+            ['<?php foo(1);', 2, 4],
+            ['<?php foo((1));', 3, 5],
+            ['<?php foo((1));', 2, 6],
+            ['<?php foo(1, 2, 3);', 2, 10],
+            ['<?php foo(1, bar(2, 3), 4);', 2, 16],
+            ['<?php $foo["bar"];', 2, 4],
+            ['<?php [1, 2, 3];', 1, 9],
+            ['<?php $foo = function ($x) { return $x + 10; };', 7, 9],
+            ['<?php $foo = function ($x) { return $x + 10; };', 11, 22],
```
Comment (SpacePossum) [inline tests/Tokenizer/Analyzer/BlocksAnalyzerTest.php]: can you add some array destructuring samples as well (in both syntax styles)?

**Comment by SpacePossum** [comment]
@kubawerlos why target master and not 2.12 here?

**Comment by kubawerlos** [comment]
@SpacePossum after all it's a new feature, right? Also I've separated the changes into few commits, so in case of bugfix for the fixer on `2.12` we can cherry-pick the first one.

**Comment by SpacePossum** [comment]
The changes to the rule/fixer are a nice improvement on the existing logic which would be great to have on 2.12. For that I'm happy to see this in 2.12, but if others think it cannot than fine as well :+1:

**Comment by dmvdbrugge** [comment]
I don't see new options, it's not an entire new fixer, I could see this as a "bugfix" (of a "forgotten" edge case) and thus targeting 2.12

**Comment by SpacePossum** [comment]
@kubawerlos would you mind to retarget to 2.12?

**Comment by kubawerlos** [comment]
@SpacePossum done

File: src/Tokenizer/Analyzer/BlocksAnalyzer.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Kuba Werłos <werlos@gmail.com>
+ *
+ * @internal
+ */
+final class BlocksAnalyzer
```
Comment (keradus) [inline src/Tokenizer/Analyzer/BlocksAnalyzer.php]: please add some docs for it, out of the blue, i don't see it as generic analyzer that deserve an separated, reusable class

File: src/Tokenizer/Analyzer/BlocksAnalyzer.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    /**
+     * @param Tokens   $tokens
+     * @param null|int $openIndex
+     * @param null|int $closeIndex
+     *
+     * @return bool
+     */
+    public function isBlock(Tokens $tokens, $openIndex, $closeIndex)
+    {
+        if (!$tokens->offsetExists($openIndex)) {
```
Comment (keradus) [inline src/Tokenizer/Analyzer/BlocksAnalyzer.php]: https://www.php.net/manual/en/splfixedarray.offsetexists.php
it expects `int` as parameter, you are giving `null|int`.

File: src/Tokenizer/Analyzer/BlocksAnalyzer.php (hunk 2 above)
Comment (kubawerlos) [reply src/Tokenizer/Analyzer/BlocksAnalyzer.php]: It's more about SRP than re-usability.

Through, I was thinking about using it when working on https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/1812

File: src/Tokenizer/Analyzer/BlocksAnalyzer.php (hunk 3 above)
Comment (kubawerlos) [reply src/Tokenizer/Analyzer/BlocksAnalyzer.php]: Yet, it there are tests for `null` and are passing.

Anyway, I've added the check for `null` to make it more clear.

**Comment by SpacePossum** [comment]
Thank you @kubawerlos.

**Pull request closed without merging**